### PR TITLE
Partially refactor JS utilities to support future webapps in separate folder structure

### PR DESCRIPTION
**Pull request by ldcWV**
## Motivation and Context

`vr_utils.js` has been added to `src/esp/bindings_js/modules`. This is a new file that contains functions that help with setting up the eye sensors and updating the sensor locations based on the user's pose.

Changes were also made to `utils.js` so that it no longer needs to import `infoSemanticFileName` from `defaults.js`. This allows it to be independently included as a module in webapps.

## How Has This Been Tested

The WebXR Hand Demo port to the new folder structure (https://github.com/facebookresearch/habitat-sim/pull/1366) which uses these utils functions was launched and ran successfully.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings_js/utils/WindowlessEmscriptenApplication.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
```
Comment (eundersander) [inline src/esp/bindings_js/utils/WindowlessEmscriptenApplication.js:1]: Magnum already provides this file, so we shouldn't add another copy in the habitat-sim repo.

File: src/esp/bindings_js/utils/utils.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (eundersander) [inline src/esp/bindings_js/utils/utils.js:1]: There's a lot of duplication here with `bindings_js/modules/utils.js`. It's confusing to have both of these in the library codebase. By "library codebase", I mean anything under habitat-sim/src, which we'll expect all webapps to share.

Can you find a way to consolidate down to just one `utils.js`? If it's not possible, it'll be better to include the version you need inside your web app (under `examples`). We shouldn't have two copies inside the library codebase.

File: src/esp/bindings_js/utils/vr_utils.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+const VIEW_SENSORS = ["left_eye", "right_eye"];
+
+function getEyeSensorSpecs() {
+  const specs = new Module.VectorSensorSpec();
+  {
+    const spec = new Module.CameraSensorSpec();
+    spec.uuid = "left_eye";
+    spec.sensorType = Module.SensorType.COLOR;
+    spec.sensorSubType = Module.SensorSubType.PINHOLE;
+    spec.resolution = [1024, 1024];
```
Comment (eundersander) [inline src/esp/bindings_js/utils/vr_utils.js:10]: 1024 probably shouldn't be hard-coded here.

File: src/esp/bindings_js/utils/vr_utils.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+const VIEW_SENSORS = ["left_eye", "right_eye"];
```
Comment (eundersander) [inline src/esp/bindings_js/utils/vr_utils.js:1]: There's a lot of duplication here with the existing `esp/bindings_js/modules/vr_demo.js`. We should discuss offline.

**Review by eundersander — COMMENTED** [review]
Let's discuss this PR more in person. I'm not sure about the best way forward.

File: src/esp/bindings_js/utils/vr_utils.js (hunk 4 above)
Comment (ldcWV) [reply src/esp/bindings_js/utils/vr_utils.js:1]: For the record: `vr_demo.js` will be deleted in https://github.com/facebookresearch/habitat-sim/pull/1366. The code duplication here is temporary.

File: src/esp/bindings_js/modules/vr_utils.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+/* global Module */
+
+export const VIEW_SENSORS = ["left_eye", "right_eye"];
+export const pointToArray = p => [p.x, p.y, p.z, p.w];
+
+export function getEyeSensorSpecs() {
+  const resWidth = 1024;
```
Comment (eundersander) [inline src/esp/bindings_js/modules/vr_utils.js:11]: I still don't think these should be hard-coded. Make them parameters so the caller code can choose.

**Review by eundersander — DISMISSED** [review]
~Please make the requested small fix but otherwise LGTM!~ Oops, actually I'd like to understand the possible code redundancy before we merge this.

File: src/esp/bindings_js/modules/vr_utils.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (eundersander) [inline src/esp/bindings_js/modules/vr_utils.js:1]: Is there any code redundancy between this file and the existing vr_demo.js?

Comment (ldcWV) [reply src/esp/bindings_js/modules/vr_utils.js:1]: Yeah, see here: https://github.com/facebookresearch/habitat-sim/pull/1364#discussion_r664908573

The existing vr_demo.js will be deleted in https://github.com/facebookresearch/habitat-sim/pull/1366, so this redundancy will be resolved as soon as that PR gets merged.

**Review by eundersander — APPROVED** [review]
LGTM

**Pull request merged**